### PR TITLE
Date picker update custom format example

### PR DESCRIPTION
**Pull request by Fercas123**
update custom format example to use DateFormat to avoid it rendering different depending on locale

### CODE REVIEW

File: .changeset/clever-shrimps-knock.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/lab": patch
+---
+
+- Remove validation logic from `DatePicker` to support integration through `FormField`.
```
Comment (Fercas123) [inline .changeset/clever-shrimps-knock.md:5]:
```suggestion
- Remove validation logic from `DatePicker` to support integration through `FormField`.
- Added `onSelect` `onSelectionChange` and `validationStatus` props to `DatePicker`.
```

File: packages/lab/src/date-picker/DatePickerPanel.tsx [hunk 2]
```diff
@@ -152,30 +154,32 @@ export const DatePickerPanel = forwardRef<HTMLDivElement, DatePickerPanelProps>(
...
           )}
           <FlexLayout>
-            <Calendar
-              visibleMonth={startVisibleMonth}
-              onVisibleMonthChange={(_, month) => setStartVisibleMonth(month)}
-              {...firstCalendarProps}
-              {...CalendarProps}
-            />
-            {isRangePicker && (
+            <FormFieldContext.Provider value={{} as FormFieldContextValue}>
```
Comment (origami-z) [inline packages/lab/src/date-picker/DatePickerPanel.tsx:157]: Not too sure why this line is here. May be a subtle edge case needs to be fixed elsewhere?

**Review by origami-z — DISMISSED** [review]

File: packages/lab/src/date-picker/DatePickerPanel.tsx (hunk 2 above)
Comment (joshwooding) [reply packages/lab/src/date-picker/DatePickerPanel.tsx:157]: We need a comment, it's to avoid the Dropdown's in Calendar inheriting the FormField's state

Comment (Fercas123) [inline packages/lab/src/date-picker/DatePickerPanel.tsx:157]:
```suggestion
            {/* Avoid Dropdowns in Calendar inheriting the FormField's state */}
            <FormFieldContext.Provider value={{} as FormFieldContextValue}>
```

**Review by origami-z — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**